ada_feeding/scripts/ada_planning_scene.py

```diff
@@ -690,24 +690,23 @@ def update_table_detection(self) -> None:
         detected_table_pose.pose.position.y += self.table_detection_offsets[1]
         detected_table_pose.pose.position.z += self.table_detection_offsets[2]
 
-        # Convert the default and latest table quaternions to pyquaternion objects
-        default_table_pyquat = Q(
-            w=self.objects[self.table_object_id].quat_xyzw[3],
-            x=self.objects[self.table_object_id].quat_xyzw[0],
-            y=self.objects[self.table_object_id].quat_xyzw[1],
-            z=self.objects[self.table_object_id].quat_xyzw[2],
+        # Calculate the angular distance between the latest and default table quaternions
+        latest_table_quat = np.array(
+            [
+                detected_table_pose.pose.orientation.x,
+                detected_table_pose.pose.orientation.y,
+                detected_table_pose.pose.orientation.z,
+                detected_table_pose.pose.orientation.w,
+            ]
         )
-        latest_table_pyquat = Q(
-            w=detected_table_pose.pose.orientation.w,
-            x=detected_table_pose.pose.orientation.x,
-            y=detected_table_pose.pose.orientation.y,
-            z=detected_table_pose.pose.orientation.z,
+        default_table_quat = np.array([0.0, 0.0, 1.0, 0.0]) # default quaternion with 180 degrees
```

Comment:
Huh, this seems strange to me for two reasons:

1. Shouldn't you be computing the difference between `latest_table_quat` and `self.objects[self.table_object_id].quat_xyzw`? I don't understand why you are introducing a new orientation here. Even if you need to rotate `self.objects[self.table_object_id].quat_xyzw` by 180 degrees, you should not hardcode the angle -- because in the future if we change the `self.objects[self.table_object_id].quat_xyzw` parameter in the YAML file, we should not need to also change this hardcoded angle in the code.
2. I don't understand why the 180 degree rotation is helpful / necessary / mathematically justified. Could you please clarify?